#! /usr/bin/env python

PACKAGE = 'fg_ros'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       name    type     level     description     default      min      max
gen.add("score_thresh", double_t, 0, "Threshold for confidence score", 0.4, 0.0, 1.0)
gen.add("nms_thresh", double_t, 0, "Threshold for nms", 0.45, 0.0, 1.0)
gen.add("max_det", int_t, 0, "Number of max detections", 1000, 1, 10000)

gen.add("roi_x_min",    int_t,    0, "ROI X min",                         0,       0,       1600)
gen.add("roi_y_min",    int_t,    0, "ROI Y min",                         0,       0,       1200)
gen.add("roi_x_max",    int_t,    0, "ROI X max",                         1920,    0,       1600)
gen.add("roi_y_max",    int_t,    0, "ROI Y max",                         1080,    0,       1200)

exit(gen.generate(PACKAGE, PACKAGE, "InstanceSegmentation"))
